internal/provider/file/file.go

```diff
@@ -45,10 +46,11 @@ func New(svr *config.Server, resources *message.ProviderResources) (*Provider, e
 	}
 
 	return &Provider{
-		paths:          paths.UnsortedList(),
-		logger:         logger,
-		watcher:        filewatcher.NewWatcher(),
-		resourcesStore: newResourcesStore(svr.EnvoyGateway.Gateway.ControllerName, resources, logger),
+		paths:             paths.UnsortedList(),
+		logger:            logger,
+		watcher:           filewatcher.NewWatcher(),
+		resourcesStore:    newResourcesStore(svr.EnvoyGateway.Gateway.ControllerName, resources, logger),
+		healthzServerPort: svr.EnvoyGateway.Provider.Custom.HealthzServerPort,
```

Comment:
we should set default port if the `HealthzServerPort` is nil

Reply:
My bad! I didn't realize the type error of it not being a pointer. Have fixed the same in the latest commit.